### ISSUE TITLE
[Bug]: UT TestUpdateWithRemotePartition failed

### ISSUE THREAD

**Issue by xzxiong** [matrixorigin/matrixone#19171]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main

### Commit ID

17e27ac2f

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

```
    partition_based_test.go:502: 
        	Error Trace:	/home/runner/work/matrixone/matrixone/pkg/tests/shard/partition_based_test.go:502
        	            				/home/runner/work/matrixone/matrixone/pkg/tests/shard/partition_based_test.go:269
        	            				/home/runner/work/matrixone/matrixone/pkg/tests/shard/shard_test.go:108
        	            				/home/runner/work/matrixone/matrixone/pkg/tests/shard/shard_test.go:40
        	            				/home/runner/work/matrixone/matrixone/pkg/tests/shard/partition_based_test.go:240
        	Error:      	Not equal: 
        	            	expected: 100
        	            	actual  : 1
        	Test:       	TestUpdateWithRemotePartition
```

![image](https://github.com/user-attachments/assets/88951ffa-1e58-423a-9792-25e0f9151789)

### Steps to Reproduce

```Markdown
https://github.com/matrixorigin/matrixone/actions/runs/11248199367/job/31272952009?pr=19168#step:9:1
```

**Comment by xzxiong** [comment]
@zhangxu19830126 ptal

**Comment by zhangxu19830126** [comment]
连续执行3个语句：

**Comment by daviszhen** [comment]
https://github.com/matrixorigin/matrixone/actions/runs/11271076832/job/31343184841?pr=19209#step:5:28740

**Comment by daviszhen** [comment]
https://github.com/matrixorigin/matrixone/actions/runs/11288756824/job/31397195941?pr=19174

**Comment by triump2020** [comment]
Pls ref to #19202

**Comment by triump2020** [comment]
UT 代码有问题

**Comment by triump2020** [comment]
The water mark to be waiting for is zero cause to this bug. @zhangxu19830126

**Comment by zhangxu19830126** [comment]
fixed by #19530
